### PR TITLE
Generalize exception type

### PR DESCRIPTION
**Pull request by timholy**
In preparation for https://github.com/JuliaLang/julia/pull/41885

### CODE REVIEW

File: test/mapreduce.jl [hunk 1]
```diff
@@ -90,10 +90,17 @@ using Statistics: mean
...
         # the error is thrown by `Base.mapreduce_empty`:
-        @test_throws(
-            ArgumentError("reducing over an empty collection is not allowed"),
-            mapreduce(nothing, nothing, SVector{0,Int}())
-        )
+        if Base.VERSION >= v"1.8.0-DEV.363"
+            @test_throws(
+                "reducing over an empty collection is not allowed",
+                mapreduce(nothing, nothing, SVector{0,Int}())
+            )
```
Comment (c42f) [inline test/mapreduce.jl:97]: For anyone (like myself) thinking this use of `@test_throws` looks confusing... see https://github.com/JuliaLang/julia/pull/41888

**Review by c42f — APPROVED** [review]
Looks good to me, pending this getting merged upstream.

**Comment by timholy** [comment]
> pending this getting merged upstream.

That's of course fine, but the fact that it returned an ArgumentError was a bit dicey from the start. One can argue that `Union{MethodError,ArgumentError}` is a better type for the error, but the new approach allows you to care only about the message itself (which seems to be what the original author of this test cared about).

**Comment by timholy** [comment]
(clicked the wrong button, sorry)

**Review by c42f — APPROVED** [review]
Ah yes, technically this only depends on https://github.com/JuliaLang/julia/pull/41888 which has already been merged. So this seems fine.

**Pull request merged**